### PR TITLE
feat(command-deploy): refresh configuration when running with --build

### PR DESCRIPTION
**Pull request by erezrokah**
**- Summary**

Fixes https://github.com/netlify/cli/issues/2501

**- Test plan**

TBD - add a new test

**- A picture of a cute animal (not mandatory but encouraged)**
🐶

### CODE REVIEW

File: tests/command.deploy.test.js [hunk 1]
```diff
@@ -348,6 +348,56 @@ if (process.env.IS_FORK !== 'true') {
     })
   })
 
+  test.only('should refresh configuration when --build is passed', async (t) => {
```
Comment (ehmicky) [inline tests/command.deploy.test.js:351]:
```suggestion
  test('should refresh configuration when --build is passed', async (t) => {
```

File: src/commands/deploy.js [hunk 2]
```diff
@@ -296,6 +296,22 @@ const runDeploy = async ({
   }
 }
 
+const handleBuild = async ({ context, flags }) => {
+  if (flags.build) {
```
Comment (ehmicky) [inline src/commands/deploy.js:300]: [sand] Should we use an early exit to avoid indentation?

File: tests/command.deploy.test.js [hunk 3]
```diff
@@ -348,6 +348,56 @@ if (process.env.IS_FORK !== 'true') {
...
+            functions: { directory: 'functions' },
+            plugins: [{ package: './plugins/mutator' }],
+          },
+        })
+        .withBuildPlugin({
+          name: 'mutator',
+          plugin: {
+            onPreBuild: async ({ netlifyConfig }) => {
+              // eslint-disable-next-line node/global-require
+              const [fs, util] = [require('fs'), require('util')]
```
Comment (ehmicky) [inline tests/command.deploy.test.js:370]: [sand] Would this work at the top-level?

**Review by ehmicky — APPROVED** [review]

File: tests/command.deploy.test.js (hunk 1 above)
Comment (erezrokah) [reply tests/command.deploy.test.js]: Nice catch!

File: tests/command.deploy.test.js (hunk 3 above)
Comment (erezrokah) [reply tests/command.deploy.test.js:370]: This translates to plugin code, so it having it top level in this file will result in missing requires in the plugin file.

File: src/commands/deploy.js (hunk 2 above)
Comment (erezrokah) [reply src/commands/deploy.js]: Done in https://github.com/netlify/cli/pull/2593/commits/21f6f2805f084201a37b5a93244022f11ae1d0d5

File: tests/command.deploy.test.js (hunk 3 above)
Comment (ehmicky) [reply tests/command.deploy.test.js:370]: :+1:

**Pull request merged**